code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Medicine.dm

```diff
@@ -1529,6 +1619,18 @@
 			if(E.status & TENDON_CUT && prob(10))
 				E.status &= ~TENDON_CUT
 
+	var/tumour_chance = 5
+	for(var/obj/item/organ/internal/parasite/malignant_tumour/T in M.internal_organs)
+		tumour_chance = max(tumour_chance-2, 0) //no more than 3 tumours
+	if(prob(tumour_chance))
+		var/obj/item/organ/external/affected = (pick(H.organs))
+		if(BP_IS_ROBOTIC(affected))
+			return
+		var/obj/item/organ/internal/parasite/malignant_tumour/infest = new()
+		infest.parent_organ = affected
+		infest.replaced(H, affected)
```

Comment:
this is copypasted a lot - make it a human proc then call it on mobs instead of copypasting it